Kernel/Memory/MemoryManager.cpp

```diff
@@ -278,152 +280,370 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map()
+                    global_data.physical_memory_ranges.append(PhysicalMemoryRange { PhysicalMemoryRangeType::Reserved, PhysicalAddress { state.start }, state.size });
+                    // FIXME: Not all of these are "used", only those in "memory" are actually "used"
+                    //        There might be for example debug DMA control registers, which are marked as reserved
+                    global_data.used_memory_ranges.append(UsedMemoryRange { UsedMemoryRangeType::BootModule, PhysicalAddress { state.start }, PhysicalAddress { state.start + state.size } });
+                    state.state = State::InReservedMemory;
+                    break;
+                }
+                state.depth--;
+                return IterationDecision::Continue;
+            },
+            .on_property = [&state](StringView property_name, ReadonlyBytes data) -> ErrorOr<IterationDecision> {
+                switch (state.state) {
+                case State::Root:
+                    if (state.depth != 1)
+                        break;
+                    if (property_name == "#address-cells"sv) {
+                        BigEndian<u32> data_as_int;
+                        __builtin_memcpy(&data_as_int, data.data(), sizeof(u32));
+                        state.address_cells = data_as_int;
+                        VERIFY(state.address_cells != 0);
+                        VERIFY(state.address_cells <= 2);
+                    } else if (property_name == "#size-cells"sv) {
+                        BigEndian<u32> data_as_int;
+                        __builtin_memcpy(&data_as_int, data.data(), sizeof(u32));
+                        state.size_cells = data_as_int;
+                        VERIFY(state.size_cells != 0);
+                        VERIFY(state.size_cells <= 2);
                     }
+                    break;
+                case State::InReservedMemory:
+                    // FIXME: We could check and verify that the address and size cells
+                    //        Are the same as in the root node
+                    break;
+                case State::InReservedMemoryChild:
+                case State::InMemory:
+                    // FIXME: Support "size" and "align" property
+                    if (property_name == "reg"sv) {
+                        VERIFY(state.address_cells);
+                        VERIFY(state.size_cells);
+                        // FIXME: We may get more than one range here
+                        if (data.size() > (state.address_cells + state.size_cells) * sizeof(u32))
+                            TODO();
+                        if (state.address_cells == 1) {
+                            BigEndian<u32> data_as_int;
+                            __builtin_memcpy(&data_as_int, data.data(), sizeof(u32));
+                            state.start = data_as_int;
+                            data = data.slice(sizeof(u32));
+                        } else {
```

Comment:
Is memory guaranteed to only use 1 or 2 address cells? Otherwise this else seems a little unsafe; you mention in the previous commit that PCI can use 3 cells.

Reply:
For these fields they should only ever be 1 or 2, as per the schema (I can add a link to that though)
PCI is a special case and has to have 3, and then uses the global(?) Address cell size for the reg field and another field

Reply:
Then there should be a comment about that.

Reply:
Added links to the schema and their consequences on the cell counts